Clojure/tests/syntax_test_clojure.clj

```diff
@@ -0,0 +1,2007 @@
+;                             ^^^^^^^^^^^^^ constant.numeric.clojure
+;                                           ^^^^^^^^^^^^^ constant.numeric.clojure
+  +1234.1234e+10M +1234.1234E+10M +1234.1234e-10M +1234.1234E-10M
+; ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                ^- constant
+;                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+  -1234.1234e+10M -1234.1234E+10M -1234.1234e-10M -1234.1234E-10M
+; ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                ^- constant
+;                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+;                                                 ^^^^^^^^^^^^^^^ constant.numeric.clojure
+
+; ## Breaks
+
+  10,20,30
+; ^^ constant.numeric.clojure
+;   ^ comment.punctuation.comma.clojure
+;    ^^ constant.numeric.clojure
+  10;20;30
+; ^^ constant.numeric.clojure
+;   ^ comment.line.clojure punctuation.definition.comment
+  10'20'30
+; ^^ constant.numeric.clojure
+;   ^ constant.other.symbol.reader-macro.clojure
+  10`20`30
+; ^^ constant.numeric.clojure
+;   ^ constant.other.symbol.reader-macro.clojure
+  10#20#30
+; ^^ constant.numeric.clojure
+;   ^ constant.other.symbol.reader-macro.clojure
+
+; ## Unaffected
+
+  '1234 '+1234 '-1234
+; ^ constant.other.symbol.reader-macro.clojure
+;  ^^^^ constant.numeric.clojure
+
+  (10 20 30) [10 20 30]
+; ^ punctuation.section.parens.begin.clojure
+;  ^^ constant.numeric.clojure
+;            ^ punctuation.section.brackets.begin.clojure
+;             ^^ constant.numeric.clojure
+
+  ([100 200])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^ constant.numeric.clojure
+;       ^^^ constant.numeric.clojure
+;          ^ punctuation.section.brackets.end.clojure
+;           ^ punctuation.section.parens.end.clojure
+  ([0x10 0x20])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^ constant.numeric.clojure
+;        ^^^^ constant.numeric.clojure
+;            ^ punctuation.section.brackets.end.clojure
+;             ^ punctuation.section.parens.end.clojure
+  ([2r100 16r200])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^^ constant.numeric.clojure
+;         ^^^^^^ constant.numeric.clojure
+;               ^ punctuation.section.brackets.end.clojure
+;                ^ punctuation.section.parens.end.clojure
+  ([10/20 30/40])
+; ^ punctuation.section.parens.begin.clojure
+;  ^ punctuation.section.brackets.begin.clojure
+;   ^^^^^ constant.numeric.clojure
+;         ^^^^^ constant.numeric.clojure
+;              ^ punctuation.section.brackets.end.clojure
+;               ^ punctuation.section.parens.end.clojure
+  ([100.100 200.200])
```

Comment:
Are these floats, or do the `.` represent something else?

Reply:
These are floats. Like many other tests, this one looks funny because it tests for funny mistakes I made when developing the syntax.